File: 3633/3633.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var earliestFinishTimeForLandAndWa = function(root) {
    if (!root) return 0;
    return 1 + Math.max(earliestFinishTimeForLandAndWa(root.left), earliestFinishTimeForLandAndWa(root.right));
};

File: 3633/3633.go
func earliestFinishTimeForLandAndWa(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := earliestFinishTimeForLandAndWa(root.Left)
	r := earliestFinishTimeForLandAndWa(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}